tests/inspection/test_lazy_wheel.py

```diff
@@ -0,0 +1,324 @@
+from __future__ import annotations
+
+import re
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+from typing import Protocol
+from typing import Tuple
+from urllib.parse import urlparse
+
+import httpretty
+import pytest
+import requests
+
+from requests import codes
+
+from poetry.inspection.lazy_wheel import HTTPRangeRequestUnsupported
+from poetry.inspection.lazy_wheel import InvalidWheel
+from poetry.inspection.lazy_wheel import metadata_from_wheel_url
+
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from httpretty.core import HTTPrettyRequest
+
+    from tests.types import FixtureDirGetter
+
+    HttPrettyResponse = Tuple[int, Dict[str, Any], bytes]  # status code, headers, body
+    HttPrettyRequestCallback = Callable[
+        [HTTPrettyRequest, str, Dict[str, Any]], HttPrettyResponse
+    ]
+
+    class RequestCallbackFactory(Protocol):
+        def __call__(
+            self,
+            *,
+            accept_ranges: str | None = "bytes",
+            negative_offset_error: tuple[int, bytes] | None = None,
+        ) -> HttPrettyRequestCallback: ...
```

Comment:
For consistency with `httpretty` and compliance with PEP8 naming:
```suggestion
    HTTPrettyResponse = Tuple[int, Dict[str, Any], bytes]  # status code, headers, body
    HTTPrettyRequestCallback = Callable[
        [HTTPrettyRequest, str, Dict[str, Any]], HTTPrettyResponse
    ]

    class RequestCallbackFactory(Protocol):
        def __call__(
            self,
            *,
            accept_ranges: str | None = "bytes",
            negative_offset_error: tuple[int, bytes] | None = None,
        ) -> HTTPrettyRequestCallback: ...
```